### ISSUE TITLE
t.mail.test.test_imap.PreauthIMAP4ClientMixin._result conflicts with Python 2.7 unittest

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3846]
|<img alt="ivank's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| ivank reported|
|-|-|
|Trac ID|trac#3846|
|Type|defect|
|Created|2009-05-20 14:21:04Z|
|Branch|https://github.com/twisted/twisted/tree/rename-private-result-3846|

Python 2.7r72791 `unittest.TestCase` has a `_result` attribute.

Summary of failures caused by the `_result` conflict:

```
  IMAP4ClientExamineTests
    test_bareOk ...                                                     [ERROR]
    test_exists ...                                                     [ERROR]
    test_flags ...                                                      [ERROR]
    test_nonIntegerExists ...                                            [FAIL]
    test_nonIntegerRecent ...                                            [FAIL]
                                          [ERROR]
    test_nonIntegerUIDNEXT ...                                           [FAIL]
    test_nonIntegerUIDVALIDITY ...                                       [FAIL]
                                     [ERROR]
                                     [ERROR]
    test_nonIntegerUnseen ...                                            [FAIL]
    test_permanentflags ...                                             [ERROR]
                                            [ERROR]
    test_recent ...                                                     [ERROR]
    test_uidnext ...                                                    [ERROR]
    test_uidvalidity ...                                                [ERROR]
    test_unrecognizedOk ...                                             [ERROR]
    test_unseen ...                                                     [ERROR]
  IMAP4ClientExpungeTests
    test_expunge ...                                                    [ERROR]
    test_nonIntegerExpunged ...                                          [FAIL]
  IMAP4ClientFetchTests
    test_fetchBody ...                                                  [ERROR]
    test_fetchSpecific ...                                              [ERROR]
    test_fetchSpecificMIME ...                                          [ERROR]
    test_fetchSpecificNumbered ...                                      [ERROR]
    test_fetchSpecificNumberedText ...                                  [ERROR]
    test_fetchSpecificPartial ...                                       [ERROR]
    test_fetchSpecificPeek ...                                          [ERROR]
    test_fetchSpecificText ...                                          [ERROR]
    test_fetchUID ...                                                   [ERROR]
    test_fetchUIDNonIntegerFound ...                                     [FAIL]
    test_incompleteFetchSpecificPartialResponse ...                      [FAIL]
    test_incompleteFetchSpecificTextResponse ...                         [FAIL]
    test_incompleteFetchUIDResponse ...                                  [FAIL]
                                [ERROR]
  IMAP4ClientSearchTests
    test_nonIntegerFound ...                                             [FAIL]
    test_search ...                                                     [ERROR]
  IMAP4ClientStoreTests
    test_addFlags ...                                                   [ERROR]
    test_addFlagsSilently ...                                           [ERROR]
    test_addFlagsSilentlyWithUnsolicitedData ...                        [ERROR]
    test_removeFlags ...                                                [ERROR]
    test_removeFlagsSilently ...                                        [ERROR]
    test_removeFlagsSilentlyWithUnsolicitedData ...                     [ERROR]
                    [ERROR]
                    [ERROR]
                    [ERROR]
                    [ERROR]
                    [ERROR]
                    [ERROR]
    test_setFlags ...                                                   [ERROR]
    test_setFlagsSilently ...                                           [ERROR]
    test_setFlagsSilentlyWithUnsolicitedData ...                        [ERROR]
```

<details><summary>Searchable metadata</summary>

```
trac-id__3846 3846
type__defect defect
reporter__ivank ivank
priority__normal normal
milestone__Python_2_7 Python-2.7
branch__branches_rename_private_result_3846 branches/rename-private-result-3846
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__easy easy
time__1242829264000000 1242829264000000
changetime__1243880136000000 1243880136000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

http://svn.python.org/view/python/trunk/Lib/unittest.py?r1=72191&r2=72219 seems to be the offending change.

We should be more careful about which private names we use, I suppose, particularly when subclassing third-party classes.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [26938]) Branching to 'rename-private-result-3846'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

(In [26939]) rename _result to _extractDeferredResult

refs [#3846](https://github.com/twisted/twisted/issues/3846)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="ivank's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_5"></a>|ivank commented|
|-|-|

Related: http://bugs.python.org/issue6072

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_6"></a>|@therve commented|
|-|-|

Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve set owner to @exarkun|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun set status to `closed`|
|-|-|

(In [26966]) Merge rename-private-result-3846

Author: exarkun
Reviewer: therve
Fixes: [#3846](https://github.com/twisted/twisted/issues/3846)

Change the name of the private `_result` helper function used by
`PreauthIMAP4ClientMixin` to `_extractDeferredResult` which both
better describes its function and which is less likely to conflict
with some other private attribute (eg, from unittest.TestCase or
from a subclass which mixes it in).

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
